### DISCUSSION TITLE
✅  webpack plugin for generating docs and frontend client

### DISCUSSION THREAD

**Discussion by Isaac-Leonard** [RobinTail/express-zod-api#715] [Ideas]
Would it be possible to create an official web pack plugin or a lone package that watches for code changes and automatically builds the openapi documentation and frontend client?
I'm happy to do this if needed

**Comment by RobinTail** [comment]
That's a nice idea. Perhaps it should be useful for those who use `webpack-dev-server`.
It's relatively clear to me the monitoring part of such plugin, however, I'm not sure how exactly it should trigger the generation in an abstract user environment.
Generating docs and a client requires a piece of code to be implemented (utilizing config and routing, that can be defined in different places of environment).
Do you have any ideas how it could be achieved?
Or it could just trigger a yarn/npm command, assuming that user should define and implement it.
I'd be glad to review some code samples.
In case you'd create such a plugin, I'd be happy to mention it in the documentation.
@Isaac-Leonard

**Reply by RobinTail** [reply]
solved in https://github.com/RobinTail/express-zod-api/discussions/771
